packages/babel-traverse/src/path/conversion.ts

```diff
@@ -286,7 +290,17 @@ function hoistFunctionEnvironment(
   if (inConstructor && superCalls.length > 0) {
     if (!allowInsertArrow) {
       throw superCalls[0].buildCodeFrameError(
-        "Unable to handle nested super() usage in arrow",
+        "When using '@babel/plugin-transform-arrow-functions', " +
+          "it's not possible to compile `super()` in an arrow function without compiling classes.\n" +
+          "Please add '@babel/plugin-transform-classes' to your Babel configuration.",
+      );
+    }
+    if (!allowInsertArrowWithRest) {
+      // NOTE: This may happen in `@babel/preset-env` with target `since 2017`, or in the default setting of `create-react-app`.
```

Comment:
```suggestion
      // preset-env with target `since 2017` enables `transform-parameters` without `transform-classes`.
```
The `create-react-app`'s default will evolve, while the target `since 2017` is almost fixed.